src/spec.c

```diff
@@ -2017,6 +2003,7 @@ static void onFlush(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent
   }
   IndexSpec_CleanAll();
   Dictionary_Clear();
+  SchemaRulesArgs_Free(ctx);
```

Comment:
Maybe call it `LegacySchemaRulesArgs_Free`?  Also I am not sure why you need it here, maybe we are hiding a problem?

Reply:
Will rename and remove from `onFlash` and we'll see what happens.